openslides_backend/migrations/migrations/0054_introduce_flexible_gender_model.py

```diff
@@ -0,0 +1,73 @@
+from collections import defaultdict
+
+from datastore.migrations import BaseModelMigration
+from datastore.shared.util import fqid_from_collection_and_id
+from datastore.writer.core import (
+    BaseRequestEvent,
+    RequestCreateEvent,
+    RequestUpdateEvent,
+)
+
+
+class Migration(BaseModelMigration):
+    """
+    This migration introduces the new gender model which enables
+    custom gender names for non default genders.
+    This requires to replace all gender strings in organization and
+    user models to be replaced with the corresponding gender id.
+    If the migration runs in memory then all gender information is set to None.
+    """
+
+    target_migration_index = 55
+
+    def migrate_models(self) -> list[BaseRequestEvent] | None:
+        events: list[BaseRequestEvent] = []
+        users = self.reader.get_all("user", ["gender"])
```

Comment:
You don't need to read the users, if it is a in-memory-migration. Therefore you read them only, if it is no in-memory-migration

Reply:
Done